pcdsdevices/state.py

```diff
@@ -593,31 +621,88 @@ class FakeTwinCATStateConfigDynamic(TwinCATStateConfigDynamic):
 fake_device_cache[TwinCATStateConfigDynamic] = FakeTwinCATStateConfigDynamic
 
 
-def get_dynamic_state_attr(state_index: int) -> str:
+def get_dynamic_state_attr(
+    state_index: int,
+    motor_index: int = 0,
+) -> str:
     """
     Get the attr string associated with a single state index.
 
     For example, the 5th state should create an attribute on
     TwinCATStateConfigDynamic called "state05". Therefore,
     get_dynamic_state_attr(5) == "state05".
 
-    This is only applicable for integers between 1
-    and the TWINCAT_MAX_STATES global variable, inclusive.
+    For a multi-dimensional states class, the motor index would
+    be nonzero and included in the attribute name. For example,
+    motor 3's state 2 should be
+    get_dynamic_state_attr(2, 3) == "m03_state02"
 
     Parameters
     ----------
     state_index : int
         The index of the state.
+    motor_index : int
+        The index of the motor. Zero has a special meaning here of
+        "there is only one motor".
 
     Returns
     -------
     state_attr : str
         The corresponding attribute name.
     """
-    return f'state{state_index:02}'
+    state_attr = f"state{state_index:02}"
+    if motor_index == 0:
+        return state_attr
+    else:
+        return f"m{motor_index}_{state_attr}"
+
+
+def state_config_dotted_attribute(
+    config_attr: str,
+    state_count: int,
+    motor_count: int = 1,
+    first_motor_only: bool = False,
+) -> list[str | None]:
```

Comment:
It looks like this is always `list[str]` now?

Reply:
This one does seem like it's only `list[str]`